### DISCUSSION TITLE
Macbook Pro Notch!

### DISCUSSION THREAD

**Discussion by pravinvekaria** [dwarvesf/hidden#177] [Ideas]
The good old Notch, some love it while others hate it!

The issue with the notch on the new MacBook Pro is that the status bar icons are being blocked by the notch; would be awesome if the team can figure out a way to sort this by having the hidden menu bar items drop down instead of expanding below the notch.

**Comment by milko** [comment]
The status bar items can become a problem, because more and more utilities only work as status bar items: this means that if you have a small laptop screen you are in trouble. The notch makes things really worse.

I was wondering if it would be possible to make an extension in which:
- An area on the right of the menu bar could be reserved to status bar items.
- This area can be adjusted in length.
- This area could be made scrollable with two small buttons at both ends of the area.
- This would allow a large number of menu bar status items, just scroll to access them.

I think such an application would be a big success, because many of these status bar applications do not (and often cannot) work in other ways, they need to be accessible all the time.

Hope you get interested ;-)

**Comment by baobaollf** [comment]
What about an approach like this. 
<img width="1141" alt="image" src="https://user-images.githubusercontent.com/55776365/153482508-d1dfb01f-477d-4d00-8119-1133d2561139.png">
Allocate some space below the menu bar and show the hidden icons there. Idea borrowed from the Bartender app.

**Reply by viettrungphan** [reply]
We can't do it, Bartender using accessibility framework to read and trigger UI contents on the screens which is can't submit to Appstore.

**Reply by baobaollf** [reply]
do you think we can do a version that won't be published to the Appstore? I would love to learn/build this feature if possible.

**Reply by dirtyhenry** [reply]
> Bartender using accessibility framework to read and trigger UI contents on the screens

Do you know how they technically do it? Would it be possible to build a branch with contents that can skip the App Store? 🤔

**Comment by milko** [comment]
I actually found Bartender after a couple of weeks, I guess it nails it.

**Comment by szbergeron** [comment]
Does anything prevent displaying the icons overflowed to the left side of the notch?

**Comment by polvoraxtreme** [comment]
It will be good to have something like bartender, or in any way to get all icons
